Automattic-Tracks-iOS/Event Logging/EventLogging+Sentry.swift

```diff
@@ -9,24 +14,34 @@ extension EventLogging {
             return
         }
 
-        /// We use the previous session's log for fatal errors because that session has presumably ended with the app crashing. If that's not the case,
-        /// the hosting app can handle that situation.
-        let logFilePath: URL?
-        switch event.level {
-            case .fatal: logFilePath = dataSource.previousSessionLogPath
-            default: logFilePath = dataSource.currentSessionLogPath
+        /// Allow the hosting app to determine the most appropriate log file to send for the error type. For example, in an application using
+        /// session-based file logging, the newest log file would be the current session, which is appropriate for debugging logs. However,
+        /// the previous session's log file is the correct one for a crash, because when the crash is sent there will already be a new log
+        /// file for the current session. Other apps may use time-based logs, in which case the same log would be the correct one.
+        ///
+        /// We also pass the timestamp for the event, as that can be useful for determining the correct log file.
+        guard let logFilePath = dataSource.logFilePath(forErrorLevel: event.errorType, at: event.timestamp) else {
+            return
         }
 
         /// Schedule the log file for upload, if available
-        if let logFilePath = logFilePath {
-            do {
-                let logFile = LogFile(url: logFilePath)
-                try enqueueLogForUpload(log: logFile)
-                event.logID = logFile.uuid
-            }
-            catch let err {
-                CrashLogging.logError(err)
-            }
+        do {
+            let logFile = LogFile(url: logFilePath)
+            try enqueueLogForUpload(log: logFile)
+            event.logID = logFile.uuid
         }
+        catch let err {
+            CrashLogging.logError(err)
+        }
+    }
+}
+
+extension Event {
+    var errorType: EventLoggingErrorType {
+        if self.level == .fatal {
+            return .fatal
+        }
+
+        return .debug
```

Comment:
I am not a fan of using `if-else` for enums because it both hides cases we currently handle and the cases that might be added after our initial implementation. The following might be more future-proof in that regard (although still not perfect), what do you think?

```suggestion
        switch self.level {
        case .fatal:
            return .fatal
        case .error, .warning, .info, .debug:
            return .debug
        @unknown default:
            return .debug
        }
```

Reply:
I originally had this as a `switch` and actually changed it (for the same reason – I normally don't like `if` statements for enums). But in this case, I really only care about the `.fatal` cases being passed handled specially – anything else, regardless of what it is should be filed under `debug` (including if they were to, say, rename `.fatal` to `.crash` or add some new type here).

I don't mind the `switch`, though it's more verbose to do the same thing. There's nothing in our style guide to address this, but your suggestion is more formally correct, so let's do that 😄.

Reply:
> But in this case, I really only care about the .fatal cases being passed handled specially

Although I knew that myself, I don't think that was 100% clear in the `if/else` version. A comment about the `if` line to indicate that probably would have been good enough, although it still wouldn't have been my personal preference.